### PR TITLE
Patch changes for clarity

### PR DESCRIPTION
**Pull request by TrySound**
Preparing to error handling issue.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -23,30 +26,31 @@ module.exports = function(options) {
...
+    if (file.isBuffer()) {
       try {
-        file = include(file, String(file.contents));
-        self.emit('data', file);
+        cb(null, include(file, String(file.contents)));
       } catch (e) {
-        self.emit('error', new gutil.PluginError('gulp-file-include', e.message));
+        cb(new PluginError(PLUGIN_NAME, e.message));
       }
     }
```
Comment (coderhaoxin) [inline index.js]: why `cb` instead of `emit`

File: index.js [hunk 2]
```diff
@@ -23,30 +26,31 @@ module.exports = function(options) {
...
-    var self = this;
-
+  function fileInclude(file, enc, cb) {
     if (file.isNull()) {
-      self.emit('data', file);
-    } else if (file.isStream()) {
+      cb(null, file);
+    }
+
+    if (file.isStream()) {
```
Comment (coderhaoxin) [inline index.js]: no need to change original `else if` style

Comment (TrySound) [inline index.js]: No need to save `this` in variable for clusure, shorter and callback is pretty node pattern

File: index.js (hunk 1 above)
Comment (TrySound) [inline index.js]: No need to save this in variable for clusure, shorter and callback is pretty node pattern

File: index.js (hunk 2 above)
Comment (TrySound) [inline index.js]: Why?
I think file could have just one state.

**Comment by coderhaoxin** [comment]
@TrySound I'll replace `es` with `through2`.

**Pull request closed without merging**